### DISCUSSION TITLE
Useful entry points that are accessible offline that may be helpful to your users.

### DISCUSSION THREAD

**Discussion by PySimpleGUI** [schlopp96/DocuPort_PSG#5] [Ideas]
As you've discovered, but perhaps not collected up like this are some "entry points" into PySimpleGUI.  I'll grab these and put somewhere useful.  You could add some entries to your table to have these called.... or not...  Suggestions are just that.  That's the nice thing about your repos.  They're yours to do with as you see fit.

Some of the entry points start with `sg.main_`.  You can use code completion to find some of them.

`sg.main` - Main test harness and gateway to built-in functionality and tests

![sg main](https://user-images.githubusercontent.com/46163555/123530086-a5e34080-d6c4-11eb-865d-c201e717d5b7.gif)

`sg.main_sdk_help()` - Window with access to the docstrings for objects (elements), functions, etc.

![image](https://user-images.githubusercontent.com/46163555/123530055-6b79a380-d6c4-11eb-955f-00bd2cc5a971.png)

`sg.main_get_debug_data()` - Window containing versions of all major system components

![image](https://user-images.githubusercontent.com/46163555/123530121-05415080-d6c5-11eb-85da-97d274164895.png)

`sg.main_open_github_issue()` - Easest interface to filing an Issue on the PySimpleGUI GitHub

![image](https://user-images.githubusercontent.com/46163555/123530125-17bb8a00-d6c5-11eb-815d-54b5e2fd83b0.png)

`sg.main_global_pysimplegui_settings()` - System settings

![image](https://user-images.githubusercontent.com/46163555/123530155-50f3fa00-d6c5-11eb-8321-4394ccbbdcc6.png)

**Comment by schlopp96** [comment]
These are all potentially great calls to implement to some degree... I'm getting some ideas! 

Also, I can't believe I haven't used `sg.main_global_pysimplegui_settings()` in past projects before lmao, that's so handy. Plus I KNOW for a fact I've read about it too under the "User API Calls" section from the documention! There's honestly just so much that I still regularly learn and discover just within PySimpleGUI alone, it's just so impressive to me.

But again dude, I can't thank you enough for taking the time to help out and give me suggestions/teach me some new things. I'm always looking out for new things from you and all the other creative folks using your farmework! Believe me, I'm always typing `pip install PySimpleGUI --upgrade` into my console lol. 

Many of my other (and some yet to be uploaded) repo's have been made using PySimpleGUI as well; I was considering linking the ones with screenshots to the wiki on your page similar to this project, if that is okay with you? I don't want to spam your repository with a ton of my half-functional crap lol!

**Comment by PySimpleGUI** [comment]
LOL... it's easier to do what I do.... call `sg.main()` and use the buttons from there.  I don't think I've ever called the global settings function.  I always go through main, the click the Global Settings button.  If you really want to live on the bleeding edge, you can click the red button and get latest latest latest code, which will overwrite your pip installed version with the one from GitHub.  You can still pip install to overwrite it.

**Comment by schlopp96** [comment]
> LOL... it's easier to do what I do.... call `sg.main()` and use the buttons from there. I don't think I've ever called the global settings function. I always go through main, the click the Global Settings button. If you really want to live on the bleeding edge, you can click the red button and get latest latest latest code, which will overwrite your pip installed version with the one from GitHub. You can still pip install to overwrite it.

Ok, the red update button is officially my new favorite way to update 😂🤣 It's so satisfying to watch  that progress meter fill up 
as the graph materializes - speaking of which is another thing I've been looking to start adding in my other repos!
